core-web/libs/portlets/dot-locales/portlet/src/lib/dot-locales-list/components/dot-locale-create-edit/DotLocaleCreateEdit.component.ts

```diff
@@ -0,0 +1,138 @@
+        DotMessagePipe,
+        InputTextModule,
+        CdkCopyToClipboard,
+        TooltipModule
+    ],
+    templateUrl: './DotLocaleCreateEdit.component.html',
+    styleUrl: './DotLocaleCreateEdit.component.scss',
+    changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class DotLocaleCreateEditComponent implements OnInit {
+    readonly config: DynamicDialogConfig<DotLocaleCreateEditData> = inject(DynamicDialogConfig);
+    private readonly dotMessageService = inject(DotMessageService);
+
+    ref = inject(DynamicDialogRef);
+    languageId: AbstractControl<string, string> | null | undefined;
+    form: FormGroup = new FormGroup({});
+    data: DotLocaleCreateEditData = this.config.data || {
+        languages: [],
+        countries: [],
+        locale: null,
+        localeList: []
+    };
+    showError = new BehaviorSubject<boolean>(false);
+    localeType: SelectItem[] = [
+        { label: this.dotMessageService.get('locale.standard.locale'), value: 1 },
+        { label: this.dotMessageService.get('locale.Custom.locale'), value: 2 }
+    ];
+
+    ngOnInit(): void {
+        this.initForm();
+    }
+
+    handleSubmit(): void {
+        if (this.form.valid) {
+            const { language, languageCode, country, countryCode } = this.form.value;
+            const isDuplicate = this.data.localeList.some(
+                (locale) => locale.isoCode?.toLowerCase() === this.getISOCode()
+            );
+
+            if (this.data.locale?.id || (!this.data.locale?.id && !isDuplicate)) {
+                this.ref.close({
+                    language,
+                    languageCode,
+                    country,
+                    countryCode,
+                    id: this.data.locale?.id
+                });
+            } else {
+                this.showError.next(true);
+            }
+        }
+    }
+
+    handleTypeChange(): void {
+        this.form.get('languageDropdown')?.setValue(null);
+        this.form.get('countryDropdown')?.setValue(null);
+    }
+
+    private initForm(): void {
+        this.form = new FormGroup({
+            language: new FormControl(this.data.locale?.language, {
+                validators: [Validators.required]
+            }),
+            languageCode: new FormControl(this.data.locale?.languageCode, {
+                validators: [Validators.required, Validators.pattern(/^[a-zA-Z0-9-]*$/)]
+            }),
+            country: new FormControl(this.data.locale?.country),
+            countryCode: new FormControl(this.data.locale?.countryCode),
+            id: new FormControl({ value: this.data.locale?.id, disabled: true }),
+            isoCode: new FormControl({ value: this.data.locale?.isoCode, disabled: true }),
+            languageDropdown: new FormControl(''),
+            countryDropdown: new FormControl('')
+        });
+
+        this.languageId = this.form.get('id');
+
+        this.form.get('languageDropdown')?.valueChanges.subscribe((language: DotISOItem) => {
```

Comment:
Move this to a private method with a self-explanatory name like `private subscribeToLanguageDropdownChanges()` and the same for `private subscribeToCountryDropdownChanges()` and remember both unsubscribe.